sopel/builtins/version.py

```diff
@@ -19,17 +19,55 @@
 GIT_DIR = os.path.join(PROJECT_DIR, '.git')
 
 
-def git_info():
-    head = os.path.join(GIT_DIR, 'HEAD')
+def _read_commit(gitdir: str, head: str) -> str | None:
+    """
+    Given paths to ``.git/`` and ``HEAD``, determine the associated commit hash
+    """
+    result = None
+
     if os.path.isfile(head):
         with open(head) as h:
             head_loc = h.readline()[5:-1]  # strip ref: and \n
-        head_file = os.path.join(GIT_DIR, head_loc)
+        head_file = os.path.join(gitdir, head_loc)
         if os.path.isfile(head_file):
             with open(head_file) as h:
                 sha = h.readline()
                 if sha:
-                    return sha
+                    result = sha
+
+    return result
+
+
+def _resolve_git_dirs(pth: str) -> tuple[str, str]:
```

Comment:
I habitually use this as a name to avoid conflict with `Path` in code where both are present, but I agree that it's more obvious in a function signature. No problem to change it here